lib/system/repr_v2.nim

```diff
@@ -0,0 +1,170 @@
+proc repr*(x: int): string {.magic: "IntToStr", noSideEffect.}
+  ## repr for an integer argument. Returns `x`
+  ## converted to a decimal string.
+
+proc repr*(x: int64): string {.magic: "Int64ToStr", noSideEffect.}
+  ## repr for an integer argument. Returns `x`
+  ## converted to a decimal string.
+
+proc repr*(x: float): string {.magic: "FloatToStr", noSideEffect.}
+  ## repr for a float argument. Returns `x`
+  ## converted to a decimal string.
+
+proc repr*(x: bool): string {.magic: "BoolToStr", noSideEffect.}
+  ## repr for a boolean argument. Returns `x`
+  ## converted to the string "false" or "true".
+
+proc repr*(x: char): string {.magic: "CharToStr", noSideEffect.}
+  ## repr for a character argument. Returns `x`
+  ## converted to a string.
+  ##
+  ## .. code-block:: Nim
+  ##   assert $'c' == "c"
+
+proc repr*(x: cstring): string {.magic: "CStrToStr", noSideEffect.}
```

Comment:
but that just gives `$x` ; IIRC repr was supposed to you you more low level debug info (eg the pointers)
```nim
proc repr2*(x: cstring): string {.magic: "CStrToStr", noSideEffect.}
var x = "abc"
var x2 = x.cstring
echo x.repr
echo x2.repr
echo x.repr2
echo x2.repr2
```
shows:
```
0x1082a0058"abc"
0x1082a0058"abc"

abc
abc
```